engine-sdk/src/env.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::{OneYoctoAttachError, PrivateCallError};
+use aurora_engine_types::account_id::AccountId;
+
+/// Timestamp represented by the number of nanoseconds since the Unix Epoch.
+pub struct Timestamp(u64);
+
+impl Timestamp {
+    pub fn new(ns: u64) -> Self {
+        Self(ns)
+    }
+
+    pub fn nanos(&self) -> u64 {
+        self.0
+    }
+
+    pub fn millis(&self) -> u64 {
+        self.0 / 1_000_000
+    }
+
+    pub fn secs(&self) -> u64 {
+        self.0 / 1_000_000_000
+    }
+}
```

Comment:
Out of curiosity, why not `Duration`?

Reply:
I looked into using `Duration`, but it did not feel appropriate. `Duration` has no fixed starting point whereas this does (namely the unix epoch). I also looked at using `SystemTime`, but that didn't feel right either because the internals are opaque and system-dependent whereas here we know the underlying raw data is always a `u64` number representing nanoseconds since the unix epoch.

Reply:
Thanks for the reasoning, makes sense to me.